pkg/apis/extensions/validation/validation.go

```diff
@@ -886,3 +887,14 @@ func ValidatePodSecurityPolicyUpdate(old *extensions.PodSecurityPolicy, new *ext
 	allErrs = append(allErrs, ValidatePodSecurityPolicySpec(&new.Spec, field.NewPath("spec"))...)
 	return allErrs
 }
+
+func ValidateSelectorImmutability(update *metav1.LabelSelector, old *metav1.LabelSelector) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if update == nil || old == nil {
```

Comment:
why is it considered ok if update == nil and old != nil?

Reply:
When an update is nil but another isn't, there should be an error.
When both updates are nil, there should be no error.

```
allErrs := field.ErrorList{}
if update == nil && old == nil {
	return allErrs
}
if !reflect.DeepEqual(update, old) { 
	allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("selector"), update, "selector must not be changed after update"))
}
return allErrs
```

Reply:
I will remove this function and use `ValidateImmutableField()` from `k8s.io/apimachinery/pkg/api/validation` as suggested by enisoc.